### PR TITLE
feat(ecocredit/marketplace): buy direct cli commands

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds 2 CLI commands for buy direct:
- - buy direct single (one order)
- - buy direct batch (multiple orders using a json file)

Closes: #983 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/client/marketplace/tx.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		where batch.json has the following form:
+		[
+			{
+			   "sell_order_id": 52,
+			   "quantity": "32.5",
+			   "bid_price": {"denom": "uregen", "amount": "32000000"},
+			   "disable_auto_retire": false,
+			   "retirement_location": "US-NY"
+			},
+		]`),
```
Comment (ryanchristo) [inline x/ecocredit/client/marketplace/tx.go:103]: We should probably be consistent with how we do batch orders in the create sell order and update sell order commands. We use yaml there but we use json here.

Any reason you chose json over yaml? Should we update here or there? cc @aleem1314

**Review by ryanchristo — COMMENTED** [review]
Looking good. Do we need to create a followup task for adding the integration tests or can we include those here? I think we can include them here without conflicting with #901 but maybe I'm missing something.

File: x/ecocredit/client/marketplace/tx.go (hunk 1 above)
Comment (technicallyty) [reply x/ecocredit/client/marketplace/tx.go:103]: ill look into adding tests as well

Comment (technicallyty) [reply x/ecocredit/client/marketplace/tx.go:103]: so looks like our CLI integration tests are still commented out until our test suite is wired up to work with ORM. i think we had this issue when we did our v3 sprint. 

should be able to add tests for this once the wiring mayhem is done

Comment (ryanchristo) [reply x/ecocredit/client/marketplace/tx.go:103]: > until our test suite is wired up to work with ORM

This should be good to go now that https://github.com/regen-network/regen-ledger/pull/901 is merged, correct?

Comment (technicallyty) [reply x/ecocredit/client/marketplace/tx.go:103]: not exactly,, https://github.com/regen-network/regen-ledger/pull/901/files#diff-d2a466a8c829cba98b8d562a0fc4193003da21c70943e4feb694c48142451612R52-R61

#901 switched the client impls to use the updated types, but the tests are not updated yet.

**Comment by technicallyty** [comment]
waiting for #1026 to add tests and merge

**Comment by ryanchristo** [comment]
Can we add a unit test for the command as it looks like we might migrate away from integration tests? #1041

**Review by ryanchristo — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**